dockertest/images.py

```diff
@@ -92,6 +92,10 @@ def __init__(self, repo, tag, long_id, created, size,
         self.user = user
 
         self.short_id = long_id[:12]
+        # docker-1.10 output includes a prefix describing the hash type
+        # FIXME: find canonical list
+        if long_id.startswith("sha256:"):
+            self.short_id = long_id[7:19]
```

Comment:
Why not extend the parsing regex with an ``(\w+:)?`` then check the length of ``len(mobj.groups())`` to see if there's a need to extract the hash method.  It could be useful to know what method is used, since it's changed before (at least once), I'll bet they change it again sometime.